File: Math Game/MGStartVC.m
//
//  MGStartVC.m
//  Math Game Test
//
//  Created by Kostya on 16.03.2018.
//  Copyright © 2018 Stolyarenko Konstantin. All rights reserved.
//

#import "MGStartVC.h"

@interface MGStartVC ()
@property (weak, nonatomic) IBOutlet UITextView *textView;
@end

@implementation MGStartVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.textView.hidden = YES;
    self.textView.alpha = 0.0;
    self.textView.clipsToBounds = YES;
    self.textView.layer.cornerRadius = 20;
    
    UISwipeGestureRecognizer *swipe = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(swipeGesture)];
    swipe.direction = UISwipeGestureRecognizerDirectionLeft;
    [self.view addGestureRecognizer:swipe];
}

- (IBAction)openRules:(id)sender
{
    if(self.textView.hidden)
    {
        self.textView.hidden = NO;
        [UIView animateWithDuration:0.5 delay:0 options:UIViewAnimationOptionTransitionFlipFromBottom
                         animations:^{
                            self.textView.alpha = 1.0;
                         } completion:^(BOOL finished) {
                         }];
    }
    else
    {
        [UIView animateWithDuration:0.5 delay:0 options:UIViewAnimationOptionTransitionFlipFromBottom
                         animations:^{
                            self.textView.alpha = 0.0;
                         } completion:^(BOOL finished) {
                             self.textView.hidden = YES;
                         }];
    }
}

-(void)swipeGesture
{
    [self performSegueWithIdentifier:@"StartPlayingSegue" sender:nil];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Math Game Swift/MGStartVC.swift
//
//  MGStartVC.swift
//  Math Game Swift
//
//  Created by Konstantin on 21.03.2018.
//  Copyright © 2018 Stolyarenko Konstantin. All rights reserved.
//

import UIKit

class MGStartVC: UIViewController
{
      @IBOutlet weak var textView: UITextView!
 
    override func viewDidLoad()
    {
        super.viewDidLoad()

        self.textView.isHidden = true
        self.textView.alpha = 0.0
        self.textView.clipsToBounds = true
        self.textView.layer.cornerRadius = 20
        
        let swipe = UISwipeGestureRecognizer(target: self, action: #selector(MGStartVC.swipeGesture))
        swipe.direction = .left
        self.view.addGestureRecognizer(swipe)
    }

    @IBAction func openRules(_ sender: Any)
    {
        if(self.textView.isHidden)
        {
            self.textView.isHidden = false
            
            UIView.transition(with: self.textView, duration: 0.5, options: .transitionCrossDissolve, animations: {
                self.textView.alpha = 1.0
            }, completion: nil)
            
        }
        else
        {
            UIView.transition(with: self.textView, duration: 0.5, options: .transitionCrossDissolve, animations: {
            self.textView.alpha = 0.0
            }, completion: { (finished: Bool) -> () in
             self.textView.isHidden = true})
        }
    }
    
    @objc func swipeGesture()
    {
        performSegue(withIdentifier:"StartPlayingSegue", sender: nil)
    }
 
    override func didReceiveMemoryWarning()
    {
        super.didReceiveMemoryWarning()
    }

}
